src/sql/workbench/contrib/notebook/test/browser/notebookEditor.test.ts

```diff
@@ -0,0 +1,184 @@
+	const untitledTextInput = instantiationService.createInstance(UntitledTextEditorInput, untitledTextEditorService.create({ associatedResource: untitledUri }));
+	const untitledNotebookInput = new UntitledNotebookInput(
+		testTitle, untitledUri, untitledTextInput,
+		undefined, instantiationService, notebookService, extensionService
+	);
+
+	const cellTextEditorGuid = generateUuid();
+	const queryTextEditor = new QueryTextEditor(
+		instantiationService.get(ITelemetryService),
+		instantiationService,
+		instantiationService.get(IStorageService),
+		instantiationService.get(ITextResourceConfigurationService),
+		instantiationService.get(IThemeService),
+		instantiationService.get(IEditorGroupsService),
+		instantiationService.get(IEditorService),
+		instantiationService.get(IConfigurationService)
+	);
+	const testNotebookEditor = new TestNotebookEditor({ cellGuid: cellTextEditorGuid, editor: queryTextEditor });
+	testNotebookEditor.id = untitledNotebookInput.notebookUri.toString();
+	testNotebookEditor.model = new NotebookModelStub();
+	notebookService.addNotebookEditor(testNotebookEditor);
+
+	setup(async () => {
+		setupPromise = new DeferredPromise<void>();
+		// Create notebookEditor
+		notebookEditor = new NotebookEditor(
+			instantiationService.get(ITelemetryService),
+			instantiationService.get(IThemeService),
+			instantiationService,
+			instantiationService.get(IStorageService),
+			new ContextViewService(instantiationService.get(ILayoutService)),
+			instantiationService.get(IKeybindingService),
+			instantiationService.get(IContextKeyService),
+			workbenchThemeService,
+			notebookService
+		);
+		let div = dom.$('div', undefined, dom.$('span', { id: 'demospan' }));
+		let parentHtmlElement = div.firstChild as HTMLElement;
+		notebookEditor.create(parentHtmlElement); // adds notebookEditor to new htmlElement as parent
+		assert.notStrictEqual(notebookEditor, undefined);
+		assert.strictEqual(notebookEditor['parent'], parentHtmlElement, 'parent of notebookEditor was not one that was expected');
+		await notebookEditor.setInput(untitledNotebookInput, EditorOptions.create({ pinned: true }));
+		setupPromise.resolve();
+	});
+
+	test('NotebookEditor-dispose: Tests that dispose() disposes all objects in its disposable store', async () => {
+		await setupPromise;
+		const mockNotebookEditor = TypeMoq.Mock.ofInstance(notebookEditor);
+		mockNotebookEditor.setup(x => x.dispose()).callback(() => notebookEditor.dispose());
+		mockNotebookEditor.object.dispose();
+		mockNotebookEditor.verify(x => x.dispose(), TypeMoq.Times.exactly(1));
+		const isDisposed = (<DisposableStore>mockNotebookEditor.object['_toDispose'])['_isDisposed'];
+		assert.ok(isDisposed, 'notebookEditor\'s disposable store must be disposed');
+	});
+
+	test('NotebookEditor-setSelection-getPosition-getLastPosition: Tests that getPosition and getLastPosition correctly return the range set by setSelection', async () => {
```

Comment:
Just remove the first part - I still don't see any value in having that. `getPosition and getLastPosition correctly return the range set by setSelection` is enough. Think of the test titles as sentences describing what is being tested.